### PR TITLE
unison: M4i -> M5b

### PR DESCRIPTION
**Pull request by ereslibre**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/unison/default.nix [hunk 1]
```diff
@@ -1,23 +1,23 @@
 { lib, stdenv, fetchurl, autoPatchelfHook
-, ncurses5, zlib, gmp
+, ncurses6, zlib, gmp
```
Comment (onny) [inline pkgs/development/compilers/unison/default.nix:2]:
```suggestion
, ncurses, zlib, gmp
```

I prefer setting ncurses version in `pks/top-level/all-packages.nix` like this:

```
  unison-ucm = callPackage ../development/compilers/unison {
    ncurses = ncurses6;
  };
```

What do you think?

Comment (ereslibre) [reply pkgs/development/compilers/unison/default.nix:2]: Are there any recommendations? In this case, if you choose the wrong ncurses, the auto patch elf hook will fail. 

In that case, IMO the visibility of the problem gets lost on the derivation and you’d have to go to all-packages to understand what the issue is.

I am not against changing if, but I just hit this issue bumping unison in this PR, and it was straightforward to identify the issue.

**Comment by res0nat0r** [comment]
I have #244253 out there and I will close it now, but a question I have is, is there a reason why we changing the release names from upstream?

`version = "1.0.${finalAttrs.milestone_id}-alpha";`

Upstream releases are named `M5b`, `M5a`, `M4i`, etc. Where as we're naming them such as `1.0M5b-alpha`.

**Comment by ereslibre** [comment]
> is there a reason why we changing the release names from upstream?

When bumping versions I have just been following the pattern, but I think we can remove the `1.0.` prefix and the `-alpha` suffix, like so:

```diff
diff --git a/pkgs/development/compilers/unison/default.nix b/pkgs/development/compilers/unison/default.nix
index 809b36ce35bd..3716f3949d63 100644
--- a/pkgs/development/compilers/unison/default.nix
+++ b/pkgs/development/compilers/unison/default.nix
@@ -7,7 +7,7 @@
 stdenv.mkDerivation (finalAttrs: {
   pname = "unison-code-manager";
   milestone_id = "M5b";
-  version = "1.0.${finalAttrs.milestone_id}-alpha";
+  version = finalAttrs.milestone_id;

   src = if (stdenv.isDarwin) then
     fetchurl {
```

**Comment by res0nat0r** [comment]
Sounds good to me. 😎 

Maybe the original authors intent was just to flag this app as alpha quality, but the release names slightly differing from the original naming scheme seems a bit confusing.

**Comment by SuperSandro2000** [comment]
According to our contributing guide this version should match the tags in this case.

**Comment by SuperSandro2000** [comment]
https://github.com/NixOS/nixpkgs/pull/245699

**Pull request merged**